cli/cmd/alert_list_fixable.go

```diff
@@ -0,0 +1,116 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/lacework/go-sdk/api"
+	"github.com/pkg/errors"
+)
+
+const remediateComponentName string = "remediate"
+
+// isRemediateInstalled returns true if the remediate component is installed
+func (c *cliState) isRemediateInstalled() bool {
+	return c.IsComponentInstalled(remediateComponentName)
+}
+
+// getTemplateIdentifiers runs the remediate component to retrieve a list
+// of remediation template identifiers
+func getRemediationTemplateIDs() ([]string, error) {
+	remediate, found := cli.LwComponents.GetComponent(remediateComponentName)
+	if !found {
+		return []string{}, errors.New("remediate component not found")
+	}
+
+	// set up environment variables
+	envs := []string{
+		fmt.Sprintf("LW_COMPONENT_NAME=%s", remediateComponentName),
+		"LW_JSON=true",
+		"LW_NONINTERACTIVE=true",
+	}
+	for _, e := range cli.envs() {
+		// don't let LW_JSON / LW_NONINTERACTIVE through here
+		if strings.HasPrefix(e, "LW_JSON=") || strings.HasPrefix(e, "LW_NONINTERACTIVE=") {
+			continue
+		}
+		envs = append(envs, e)
+	}
+	stdout, stderr, err := remediate.RunAndReturn([]string{"ls", "templates"}, nil, envs...)
+	if err != nil {
+		cli.Log.Debugw("remediate error details", "stderr", stderr)
+		return []string{}, err
+	}
+
+	var templates []map[string]interface{}
+	err = json.Unmarshal([]byte(stdout), &templates)
+	if err != nil {
+		return []string{}, err
+	}
+
+	templateIDs := []string{}
+	for _, template := range templates {
+		v, ok := template["id"]
+		if !ok {
+			continue
+		}
+		s, ok := v.(string)
+		if !ok {
+			continue
+		}
+		templateIDs = append(templateIDs, s)
+	}
+	return templateIDs, nil
+}
+
+// filterFixableAlerts identifies which alerts have corresponding remediation template IDs
+// and returns those which don't
+func filterFixableAlerts(
+	alerts api.Alerts,
+	getIDsFunc func() ([]string, error),
```

Comment:
any reason I am missing for you to pass a function here instead of calling it and passing just the `[]string`?

Reply:
This makes it much easier to unit test

Reply:
I see what you mean...will fix.

Reply:
Fixed.